src/opentimelineio/clip.h

```diff
@@ -9,10 +9,18 @@ namespace opentimelineio { namespace OPENTIMELINEIO_VERSION {
 class Clip : public Item
 {
 public:
+    struct MediaRepresentation
+    {
+        static char constexpr default_media[]         = "DEFAULT_MEDIA";
+        static char constexpr high_resolution_media[] = "HIGH_RESOLUTION_MEDIA";
+        static char constexpr proxy_resolution_media[] =
+            "PROXY_RESOLUTION_MEDIA";
+    };
```

Comment:
Hi @rogernelson!

I'm sorry if this is obvious in the code, but will users be able to assign their own keys or is this going to be _the_ list of available keys?

May I also suggest we consider something like "high_quality_media" and "low/proxy_quality_media" instead of "*_resolution_media"? Codecs considered "proxy" may be of high resolution (At least in the pixel dimensions sense).

Reply:
There's nothing preventing users from using their own strings.  I modelled it after the Marker::Color, so there's a list of pre-defined strings to use, but any string they pass in is going to work -- unless we wanted to prevent that, but I don't think we do.

Reply:
I definitely don't mind `quality` over `resolution`.

Reply:
Nice! 
I totally agree with leaving it open for users to add their own :)

Reply:
The risk of using something not in the list is basically interchange between different applications. Apps can easily add support for whatever is in the pre-defined list, but they can't guess at all the different strings the users could put in there, so if you do use your own strings it might not be understood outside of your own context (which is of course fine if you only intended to use the otio file within your own context).

Reply:
Potentially, but I imagine it could be possible to have high/proxy quality media available both locally and in the cloud.  I was hoping to get more suggestions for the list here anyway, so I could change the list to `disk_high_quality`, `disk_proxy_quality`, `cloud_high_quality`, `cloud_proxy_quality` if that makes sense.

Reply:
Hot take: What if we didn't have a "standard" set of keys defined. The representations that will be available and the heuristic for choosing the appropriate one will be highly pipeline/use-case dependent. I think having "standard" keys encourages trying to encode too much meaning in them and the set will infinitely grow with use cases - it'll become a really unmanageable list. I think it's out of scope for OpenTimelineIO to be trying to define workflow meanings for representations.

In a lot of ways this lines up with adaptive streaming. In that case a player chooses the appropriate representation using an algorithm that takes into account a user's measured connection speed, characteristics of the display device (resolution, bit depth, etc.), how important it is for playback to start immediately vs. prioritizing quality, supported codecs, whether or not there is time to pre-load, etc. The manifest [encodes relevant unopinionated info](https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-4.4.6.2) to feed that decision process, but it's up to the player to decide what is important based on the context.

What if, instead, we thought of the keys as being a simple human-friendly selector that can be chosen based on what makes sense for the workflow. Then, for the more advanced programmatic heuristics we could encourage the use of metadata on the media reference.

If there are other bits of meaning we want to encode to aid in choosing an appropriate rep, we should talk about ways to explicitly encode those.

Reply:
I have to say I'm biased in favor of the "hot take" no standard set of keys. I think setting active, and having a rule for what a viewer should display in the case there is no active, should suffice.

Reply:
Alright, I'm okay with it too. I think the one exception being 'default_media', correct? When we convert a Clip.1 to a Clip.2 we will need to set its key to something, and we can also use this value and the default argument in the ctor.

Reply:
Good call, it would probably be a good idea to standardize the "default" key. Over time other standardized keys may also emerge, it'll be a good thing to check in with the community about.

On the point of providing intended use case examples - I think that's a reasonable point. Perhaps we can address it with some good code/sample file examples. Additionally, we can look at adding some more documentation to clarify intended use cases.